src/Console/Command/ExportProducts.php

```diff
@@ -0,0 +1,139 @@
+<?php
+declare(strict_types=1);
+
+namespace Omikron\Factfinder\Console\Command;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Framework\App\State;
+use Magento\Framework\Filesystem;
+use Omikron\Factfinder\Model\Api\PushImport;
+use Omikron\Factfinder\Model\Export\FeedFactory as FeedGeneratorFactory;
+use Omikron\Factfinder\Model\FtpUploader;
+use Omikron\Factfinder\Model\StoreEmulation;
+use Omikron\Factfinder\Model\Stream\CsvFactory;
+use Omikron\Factfinder\Model\Config\CommunicationConfig;
+
+class ExportProducts extends \Symfony\Component\Console\Command\Command
+{
+    /** @var ScopeConfigInterface */
+    private $scopeConfig;
+
+    /** @var StoreEmulation */
+    private $storeEmulation;
+
+    /** @var FeedGeneratorFactory */
+    private $feedGeneratorFactory;
+
+    /** @var StoreManagerInterface */
+    private $storeManager;
+
+    /** @var CommunicationConfig */
+    private $communicationConfig;
+
+    /** @var CsvFactory */
+    private $csvFactory;
+
+    /** @var FtpUploader */
+    private $ftpUploader;
+
+    /** @var string */
+    private $feedType;
+
+    /** @var PushImport */
+    private $pushImport;
+
+    /** @var State */
+    private $state;
+
+    /** @var Filesystem */
+    private $filesystem;
+
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        StoreManagerInterface $storeManager,
+        FeedGeneratorFactory $feedFactory,
+        StoreEmulation $emulation,
+        CsvFactory $csvFactory,
+        FtpUploader $ftpUploader,
+        CommunicationConfig $communicationConfig,
+        PushImport $pushImport,
+        State $state,
+        Filesystem $filesystem,
+        string $name = null)
+    {
+        $this->scopeConfig = $scopeConfig;
+        $this->storeManager = $storeManager;
+        $this->feedGeneratorFactory = $feedFactory;
+        $this->storeEmulation = $emulation;
+        $this->csvFactory = $csvFactory;
+        $this->ftpUploader = $ftpUploader;
+        $this->communicationConfig = $communicationConfig;
+        $this->pushImport = $pushImport;
+        $this->state = $state;
+        $this->filesystem = $filesystem;
+        parent::__construct($name);
```

Comment:
Please call parent **before** doing anything else in the constructor.

The parameter $name here is basically useless, because it is overwritten immediately in the `configure()` method. 
Personally, I would prefer calling `parent::__construct();` and removing the constructor parameter.

Reply:
I changed it.